### DISCUSSION TITLE
Sharing Domain class between Bounded Context

### DISCUSSION THREAD

**Discussion by mehdihadeli** [kgrzybek/modular-monolith-with-ddd#181] [Q&A]
Hi @kgrzybek,
In this project you don't use a `shared kernel` component for sharing domain concepts between `bounded contexts` and in each module you created a duplicate domain concept. 
For example in `Payment Module` you created a `UserId` in payment domain again instead of moving this to a `shared kernel` component to sharing between all modules.
Which approach is better?
Maybe duplicating domain concepts in each module makes transition to microservices easier, correct?

**Comment by leandro-almeida** [comment]
Hi, from the Readme.md he states that one module cant communicate ou directly access others. Only through events.
Other reason is that UserId on Payment Module is stored on payment schema. The UserId on another module is stored in another schema, so in essence they arent the same domain object that can be shared. 
Thats is my opinion.
